include/os/linux/zfs/sys/zfs_znode_impl.h

```diff
@@ -36,20 +36,25 @@
 #include <sys/list.h>
 #include <sys/dmu.h>
 #include <sys/sa.h>
+#include <sys/time.h>
 #include <sys/zfs_vfsops.h>
 #include <sys/rrwlock.h>
 #include <sys/zfs_sa.h>
 #include <sys/zfs_stat.h>
 #include <sys/zfs_rlock.h>
 
-
 #ifdef	__cplusplus
 extern "C" {
 #endif
 
+#ifdef STATX_BTIME
```

Comment:
Since ZFS has had crtime forever I think it would be pretty reasonable to unconditionally add `z_btime` to the znode.  Then in `zfs_znode_alloc()` and `zfs_rezget()` we could do away with the ugly ifdefs.  It's only in `zpl_getattr_impl()` where it's possible the we're running an old kernel and thus there's no place to set this.  This It's true this would waste some space for 4.11 and older kernels, but these kernels are quite old now and I don't think it would be a significant issue.